src/cloud-providers/gcp/provider.go

```diff
@@ -0,0 +1,191 @@
+			return nil, fmt.Errorf("configuration error when using creds: %s", err)
+		}
+		provider.instancesClient, err = compute.NewInstancesRESTClient(context.TODO(), option.WithCredentials(creds))
+		if err != nil {
+			return nil, fmt.Errorf("NewInstancesRESTClient with credentials error: %s", err)
+		}
+	} else {
+		var err error
+		provider.instancesClient, err = compute.NewInstancesRESTClient(context.TODO())
+		if err != nil {
+			return nil, fmt.Errorf("NewInstancesRESTClient error: %s", err)
+		}
+	}
+	return provider, nil
+}
+
+func getIPs(instance *computepb.Instance) ([]netip.Addr, error) {
+	var podNodeIPs []netip.Addr
+	for _, nic := range instance.GetNetworkInterfaces() {
+		for _, access := range nic.GetAccessConfigs() {
+			ipStr := access.GetNatIP()
+			ip, err := netip.ParseAddr(ipStr)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse pod node IP %q: %w", ipStr, err)
+			}
+			podNodeIPs = append(podNodeIPs, ip)
+			logger.Printf("%s", ip.String())
+		}
+	}
+	return podNodeIPs, nil
+}
+
+func (p *gcpProvider) CreateInstance(ctx context.Context, podName, sandboxID string, cloudConfig cloudinit.CloudConfigGenerator, spec provider.InstanceTypeSpec) (*provider.Instance, error) {
+
+	instanceName := util.GenerateInstanceName(podName, sandboxID, maxInstanceNameLen)
+	logger.Printf("CreateInstance: name: %q", instanceName)
+
+	userData, err := cloudConfig.Generate()
+	if err != nil {
+		return nil, err
+	}
+
+	//Convert userData to base64
+	userDataEnc := base64.StdEncoding.EncodeToString([]byte(userData))
+	logger.Printf("userDataEnc:  %s", userDataEnc)
+
+	insertReq := &computepb.InsertInstanceRequest{
+		Project: p.serviceConfig.ProjectId,
+		Zone:    p.serviceConfig.Zone,
+		InstanceResource: &computepb.Instance{
+			Name: proto.String(instanceName),
+			Disks: []*computepb.AttachedDisk{
+				{
+					InitializeParams: &computepb.AttachedDiskInitializeParams{
+						DiskSizeGb:  proto.Int64(20),
+						SourceImage: proto.String(fmt.Sprintf("projects/%s/global/images/%s", p.serviceConfig.ProjectId, p.serviceConfig.ImageName)),
+						DiskType:    proto.String(fmt.Sprintf("zones/%s/diskTypes/pd-standard", p.serviceConfig.Zone)),
+					},
+					AutoDelete: proto.Bool(true),
+					Boot:       proto.Bool(true),
+					Type:       proto.String(computepb.AttachedDisk_PERSISTENT.String()),
+				},
+			},
+			Metadata: &computepb.Metadata{
+				Items: []*computepb.Items{
+					{
+						Key:   proto.String("user-data"),
+						Value: proto.String(userDataEnc),
+					},
+					{
+						Key:   proto.String("user-data-encoding"),
+						Value: proto.String("base64"),
+					},
+				},
+			},
+			MachineType: proto.String(fmt.Sprintf("zones/%s/machineTypes/%s", p.serviceConfig.Zone, p.serviceConfig.MachineType)),
+			NetworkInterfaces: []*computepb.NetworkInterface{
+				{
+					AccessConfigs: []*computepb.AccessConfig{
+						{
+							Name:        proto.String("External NAT"),
+							NetworkTier: proto.String("PREMIUM"),
```

Comment:
When I read "premium" my brain automatically associates with "expensive". So it would be nice to have a option for users to tune this.

Reply:
lol, makes sense. I changed the default to STANDARD. But I will create an issue for future fine tuning.